common/lru_policy.go

```diff
@@ -0,0 +1,118 @@
+package common
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+)
+
+type CacheBlock struct {
+	sync.RWMutex
+	StartIndex int64
+	EndIndex   int64
+	Data       []byte
+	Last       bool //last block in the file?
+}
+
+//KeyPair: the list node containing both block key and cache block values
+type KeyPair struct {
+	key   int64
+	value *CacheBlock
+}
+
+//LRUCache definition for Least Recently Used Cache implementation
+type LRUCache struct {
+	sync.RWMutex
```

Comment:
class holding this cache is also having a lock, we can remove one as they are duplicates